File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.java
class Solution {
    public int countPairs(List<Integer> nums, int target) {
        
        nums.sort(null);
        int l = 0;
        int r = nums.size() -1;
        int c = 0;
        while(l < r){
            if(nums.get(l) + nums.get(r) < target){
                c+= (r-l);
                l++;
            } else{
                r--;
            }
        }
        return c;
        
    }
}

File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.py
class Solution:
    def countPairs(self, nums: List[int], target: int) -> int:
        
        if(len(nums) == 0):
            return 0
        
        nums = sorted(nums)
        l = 0
        r = len(nums) -1
        c = 0
        
        while l < r:
            if nums[l] + nums[r] < target:
                c+=r-l
                l+=1
            else:
                r-=1
        
        return c